### PR TITLE
feat: filter boards by team endpoint

### LINKED ISSUE #756: [FEATURE]: Filter boards by team
## Summary

When user filters boards by team, they should see the boards that belong to that specific team (with lazy loading).

### Motivation

Team members should be able to filter all the boards by team to facilitate visualisation.

### PR DESCRIPTION
**Pull request by patricia-mdias**
Relates to #756

## Proposed Changes

  - Filter boards by team endpoint (with pagination)

This pull request closes #756

### CODE REVIEW

File: backend/src/modules/boards/controller/boards.controller.ts [hunk 1]
```diff
@@ -134,9 +134,16 @@ export default class BoardsController {
...
 	})
 	@Get()
-	async getAllBoards(@Req() request: RequestWithUser, @Query() { page, size }: PaginationParams) {
+	async getAllBoards(
+		@Req() request: RequestWithUser,
+		@Query() { page, size, team }: PaginationParams
+	) {
 		const { _id: userId, isSAdmin } = request.user;
 
+		if (team) {
```
Comment (rpvsilva) [inline backend/src/modules/boards/controller/boards.controller.ts:143]: I know that this code was already here, but can you create a `getAllBoards` on board application and make this logic there? Then this controller just return the result from that method

File: backend/src/modules/boards/services/get.board.service.ts [hunk 2]
```diff
@@ -79,6 +79,14 @@ export default class GetBoardServiceImpl implements GetBoardServiceInterface {
 		return this.getBoards(false, query, page, size);
 	}
 
+	async getTeamBoards(teamId: string, page: number, size?: number) {
```
Comment (rpvsilva) [inline backend/src/modules/boards/services/get.board.service.ts:82]:
```suggestion
	getTeamBoards(teamId: string, page: number, size?: number) {
```

File: frontend/src/api/boardService.tsx [hunk 3]
```diff
@@ -39,9 +39,25 @@ export const getDashboardBoardsRequest = (
 
 export const getBoardsRequest = (
   pageParam: number,
```
Comment (rpvsilva) [inline frontend/src/api/boardService.tsx:41]:
```suggestion
  pageParam: = 0,
```

File: frontend/src/api/boardService.tsx [hunk 4]
```diff
@@ -39,9 +39,25 @@ export const getDashboardBoardsRequest = (
...
+  team?: string,
   context?: GetServerSidePropsContext,
 ): Promise<{ boards: BoardType[]; hasNextPage: boolean; page: number }> =>
-  fetchData(`/boards?page=${pageParam ?? 0}&size=10`, { context, serverSide: !!context });
+  fetchData(
+    team
+      ? `/boards?team=${team}&page=${pageParam ?? 0}&size=10`
+      : `/boards?page=${pageParam ?? 0}&size=10`,
```
Comment (rpvsilva) [inline frontend/src/api/boardService.tsx:48]:
```suggestion
      ? `/boards?team=${team}&page=${pageParam}&size=10`
      : `/boards?page=${pageParam}&size=10`,
```

**Review by rpvsilva — CHANGES_REQUESTED** [review]

**Review by rpvsilva — APPROVED** [review]
Great job! 🚀

**Comment by patricia-mdias** [comment]
I don't understand why the frontend build fails, since I didn't change the files the build is complaining. However, I checked those files, ran the build locally and everything looks fine.

**Comment by rpvsilva** [comment]
I've just rebased with the main and fixed the issue, the problem was with the last commit to the main.

**Review by rpvsilva — APPROVED** [review]

**Pull request merged**